pkg/datasource/datasource.go

```diff
@@ -49,71 +88,64 @@ func newStravaDatasource(dsInfo *datasource.DatasourceInfo, dataDir string) (*St
 	}, nil
 }
 
-func (p *StravaPlugin) NewStravaDatasource(dsInfo *datasource.DatasourceInfo) (*StravaDatasource, error) {
-	ds, err := newStravaDatasource(dsInfo, p.dataDir)
+// getDSInstance Returns cached datasource or creates new one
+func (ds *StravaDatasource) getDSInstance(pluginContext backend.PluginContext) (*StravaDatasourceInstance, error) {
+	instance, err := ds.im.Get(pluginContext)
 	if err != nil {
 		return nil, err
 	}
-
-	ds.logger = p.logger
-	return ds, nil
+	return instance.(*StravaDatasourceInstance), nil
 }
 
-// Query receives requests from the Grafana backend. Requests are filtered by query type and sent to the
-// applicable StravaDatasource.
-func (p *StravaPlugin) Query(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (resp *datasource.DatasourceResponse, err error) {
-	StravaDS, err := p.GetDatasource(tsdbReq)
-	if err != nil {
-		return nil, err
-	}
-
-	queryType, err := getQueryType(tsdbReq)
-	if err != nil {
-		return nil, err
-	}
+func (ds *StravaDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	qdr := backend.NewQueryDataResponse()
 
-	switch queryType {
-	case StravaApiQueryType:
-		resp, err = StravaDS.StravaAPIQuery(ctx, tsdbReq)
-	case StravaAuthQueryType:
-		resp, err = StravaDS.StravaAuthQuery(ctx, tsdbReq)
-	default:
-		err = errors.New("Query not implemented")
-		return buildErrorResponse(err, nil), nil
+	for _, q := range req.Queries {
+		res := backend.DataResponse{}
+		res.Error = ErrAlertingNotSupported
```

Comment:
Why are you always setting this error?

Reply:
Because plugin doesn't support alerting and error should be in each query response.